types/src/json/block.rs

```diff
@@ -0,0 +1,299 @@
+//! JSON block types.
+
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+use crate::json::output::Output;
+
+/// JSON representation of a block.
+///
+/// Used in:
+/// - [`/get_block` -> `json`](https://www.getmonero.org/resources/developer-guides/daemon-rpc.html#get_block)
+#[derive(Clone, Default, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct Block {
+    pub major_version: u8,
+    pub minor_version: u8,
+    pub timestamp: u64,
+    pub prev_id: String,
+    pub nonce: u32,
+    pub miner_tx: MinerTransaction,
+    pub tx_hashes: Vec<String>,
+}
+
+/// [`Block::miner_tx`].
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[serde(untagged)]
+pub enum MinerTransaction {
+    V1 {
+        /// This field is [flattened](https://serde.rs/field-attrs.html#flatten).
+        #[serde(flatten)]
+        prefix: MinerTransactionPrefix,
+        signatures: [(); 0],
+    },
+    V2 {
+        /// This field is [flattened](https://serde.rs/field-attrs.html#flatten).
+        #[serde(flatten)]
+        prefix: MinerTransactionPrefix,
+        rct_signatures: MinerTransactionRctSignatures,
+    },
+}
```

Comment:
A separate type exists for miner transactions because:

- Fields are slightly different
- The `Input` JSON is different
- It allows us to enforce `signature` always being empty in miner transactions